### ISSUE TITLE
Non used override requirements brings error in CMake

### ISSUE THREAD

**Issue by climblinne** [conan-io/conan#5680]
To help us debug your issue please explain:

- [ x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [ x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [ x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

I use conan 17.2 on a Windows 10 computer.

Our Jenkins / Gerrit server delivers a list of changed libraries, which are in review phase.
So e.g. I got `LibA` and `Lib B`. Both are changed.
`LibC` depends on `LibA`.
`LibD` depends on `LibA` and `LibB`.
I generate an "override.txt", where "LibA" and "LibB" from the review channel are inside.

Now I can build `LibD` without a problem, because it uses both libraries (`LibA` & `LibB`).

The `test_package` of `LibC` will not build because the generated `conanbuildinfo.cmake` has the following dependency:

```
conan_package_library_targets("${CONAN_LIBS_LIBC}" "${CONAN_LIB_DIRS_LIBC}"
                                  CONAN_PACKAGE_TARGETS_LIBC "CONAN_PKG::LibA CONAN_PKG::LibB" "" LibC)
```
Where there is no definition inside the cmake file, so that he can't find target:

```
CMake Error at CMakeLists.txt:12 (add_executable):
  Target "LibC-test" links to target
  "CONAN_PKG::LibB" but the target was not found.  Perhaps
  a find_package() call is missing for an IMPORTED target, or an ALIAS target
  is missing?
```

Should this behave in this way, when there is an unused "override requirement"?
Is there a good way to detect the not used "override requirements" before (e.g. `conan info`)?

**Comment by memsharded** [comment]
Hi @climblinne 

Could you please clarify how you are using that ``override.txt`` file? What packages use it? How? Are you explicitly using the ``override`` qualifier in the recipe? Please let me know a bit more to understand what you are trying to achieve. Thanks.

**Comment by climblinne** [comment]
@memsharded : Here my excerpt from my base class for libraries:

```python
def requirements(self):
        try:
            with open("./overrides.txt") as file:  
                overrides = file.read()
        except FileNotFoundError:
            print("Unable to find 'overrides.txt' in {}".format(os.getcwd()))
            overrides = None            
        if overrides:
            target = overrides.split(":")
            if len(target[0]) > 0:
                overrides = target[0].split(",")
            else:
                overrides = []
               
            for override in overrides:
                ref = ConanFileReference.loads(override)
                if ref.name == self.name:
                    print("Override ignored because own package:     " + str(override) + " in " + self.name)
                elif ref.name in self.requires:
                    print("Override removed the following reference: " + str(self.requires[ref.name]))
                    print("           Replaced it by this reference: " + str(override))
                    del(self.requires[ref.name])
                    #TODO: Check if version is correct ref.version
                    self.requires(override, override=False)
                else:
                    print("Override added:                           " + str(override) + " in " + self.name)
                    self.requires(override, override=True)
```

Sample `overrides.txt`:
```
LibA/1.0.0-dev.7+HEAD.g32faf5a@review/17384.7,LibB/1.0.0-dev.10+HEAD.g2e52ca5@review/17550.1
```

Our gerrit server informs us about change of some libraries. Now Jenkins starts building these libraries (in correct order uses "dependencies"). They are named with `<library>/<version>@review/123.4`.
After theses it starts all projects, that are dependent on one of these projects in parallel. It uses the `override.txt` filled with the changed libraries. It either substitute the requirement or write an overwrite requirement.

I hope this helps a little bit.

**Comment by climblinne** [comment]
So now  I investigate more in this bug.
A demo project could be found at https://github.com/climblinne/issue-5680.

The main issue is the following:
1. Build library `HelloC`.
2. Build library `HelloB` with an **override** dependency to `HelloC`.
3. Build executable `HelloA` with an dependency to `HelloB` and an override dependency to `HelloC`.

Now with Conan 1.19.2 I get the following output:

```
HelloA/0.1@user/testing: Building your package in C:\.conan\data\HelloA\0.1\user\testing\build\a8c031b0151521648fa249b62f2a654a9211671b
HelloA/0.1@user/testing: ERROR: Generator cmake(file:conanbuildinfo.cmake) failed
'HelloC'
ERROR: 'HelloC'
```

I already attached a patch for the issue.

**Comment by memsharded** [comment]
The fix was backported to 1.19.3, which will be released soon.
